spec/mongoid/criteria_spec.rb

```diff
@@ -3047,6 +3062,20 @@ class D
         end
       end
     end
+
+    context 'when plucking a field to be demongoized' do
+      with_config_values(:legacy_pluck_distinct, true, false) do |val|
+        let(:plucked) do
+          Band.where(name: maniacs.name).pluck(:sales)
+        end
+
+        it "demongoizes the field" do
+          res = val ? "1E2" : BigDecimal("1E2")
```

Comment:
This also needs to be expanded to two cases with the correct description for each case.